modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedNode.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.discovery.isolated;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.ignite.cache.CacheMetrics;
+import org.apache.ignite.cluster.ClusterMetrics;
+import org.apache.ignite.internal.ClusterMetricsSnapshot;
+import org.apache.ignite.internal.managers.discovery.IgniteClusterNode;
+import org.apache.ignite.lang.IgniteProductVersion;
+
+import static org.apache.ignite.internal.IgniteNodeAttributes.ATTR_NODE_CONSISTENT_ID;
+
+/**
+ * Special isolated node.
+ */
+public class IsolatedNode implements IgniteClusterNode {
+    /** */
+    private final UUID id = UUID.randomUUID();
+
+    /** Consistent ID. */
+    private Object consistentId;
+
+    /** Node attributes. */
+    private Map<String, Object> attrs;
+
+    /** */
+    private IgniteProductVersion ver;
+
+    /** */
+    private ClusterMetrics metrics = new ClusterMetricsSnapshot();
+
+    /** */
+    private Map<Integer, CacheMetrics> cacheMetrics = new HashMap<>();
+
+    /** {@inheritDoc} */
+    @Override public UUID id() {
+        return id;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Object consistentId() {
+        return consistentId;
+    }
+
+    /** {@inheritDoc} */
+    @Override public <T> T attribute(String name) {
+        return (T)attrs.get(name);
+    }
+
+    /** {@inheritDoc} */
+    @Override public ClusterMetrics metrics() {
+        return metrics;
+    }
+
+    /** {@inheritDoc} */
+    @Override public Map<String, Object> attributes() {
+        return attrs;
+    }
+
+    /**
+     * @param attrs Node attributes.
+     */
+    public void attributes(Map<String, Object> attrs) {
+        this.attrs = attrs;
```

Comment:
attrs should be unmodifiable, look at org.apache.ignite.spi.discovery.tcp.internal.TcpDiscoveryNode#setAttributes

Reply:
fixed